### DISCUSSION TITLE
查询假死，状态一直为running，请大佬帮忙看下

### DISCUSSION THREAD

**Discussion by larryju** [apache/doris#8164] [A - General / Q&A]
1. 执行某些需要读取大量数据的查询时，状态一直为running，既不报错，也不会完成。
2. 查询语句只是简单的 select max(id) from table where id > xxx;
3. 分析query profile, 发现有4个节点的OLAP_SCAN_NODE 没有数据输出 
4. query profile 如下：
     `Query:
    Summary:
          -  Query  ID:  f9255570a33e421c-9072a386f7446d5e
          -  Start  Time:  2022-02-21  15:41:01
          -  End  Time:  N/A
          -  Total:  3m51s
          -  Query  Type:  Query
          -  Query  State:  RUNNING
          -  Doris  Version:  0.15.0-rc04
          -  User:  root
          -  Default  Db:  default_cluster:test
          -  Sql  Statement:  /*  ApplicationName=DBeaver  Enterprise  21.2.0  -  SQLEditor    */  select  max(id)  from  ods_count_advertising_log
where  id  >  936654880
LIMIT  0,  200
          -  Is  Cached:  No
        Execution  Summary:
              -  Analysis  Time:  339.623us
              -  Plan  Time:  378.342us
              -  Schedule  Time:  2.27ms
              -  Wait  and  Fetch  Result  Time:  N/A
    Execution  Profile  f9255570a33e421c-9072a386f7446d5e:(Active:  3m51s,  %  non-child:  100.00%)
        Fragment  0:
            Instance  f9255570a33e421c-9072a386f7446d64  (host=TNetworkAddress(hostname:172.18.65.4,  port:9060)):(Active:  3m51s,  %  non-child:  0.00%)
                  -  FragmentCpuTime:  59.866us
                  -  MemoryLimit:  2.00  GB
                  -  PeakMemoryUsage:  28.01  KB
                  -  PeakReservation:  0.00  
                  -  PeakUsedReservation:  0.00  
                  -  RowsProduced:  0
                BlockMgr:
                      -  BlockWritesOutstanding:  0
                      -  BlocksCreated:  0
                      -  BlocksRecycled:  0
                      -  BufferedPins:  0
                      -  BytesWritten:  0.00  
                      -  MaxBlockSize:  8.00  MB
                      -  TotalBufferWaitTime:  0ns
                      -  TotalEncryptionTime:  0ns
                      -  TotalIntegrityCheckTime:  0ns
                      -  TotalReadBlockTime:  0ns
                DataBufferSender  (dst_fragment_instance_id=f9255570a33e421c-9072a386f7446d64):
                      -  AppendBatchTime:  0ns
                          -  ResultSendTime:  0ns
                          -  TupleConvertTime:  0ns
                      -  NumSentRows:  0
                AGGREGATION_NODE  (id=3):(Active:  3m51s,  %  non-child:  0.00%)
                      -  Probe  Method:  HashTable  Linear  Probing
                      -  BuildTime:  0ns
                      -  GetResultsTime:  0ns
                      -  HTResize:  0
                      -  HTResizeTime:  0ns
                      -  HashBuckets:  0
                      -  HashCollisions:  0
                      -  HashFailedProbe:  0
                      -  HashFilledBuckets:  0
                      -  HashProbe:  0
                      -  HashTravelLength:  0
                      -  LargestPartitionPercent:  0
                      -  MaxPartitionLevel:  0
                      -  NumRepartitions:  0
                      -  PartitionsCreated:  0
                      -  PeakMemoryUsage:  20.00  KB
                      -  RowsProcessed:  0
                      -  RowsRepartitioned:  0
                      -  RowsReturned:  0
                      -  RowsReturnedRate:  0
                      -  SpilledPartitions:  0
                    EXCHANGE_NODE  (id=2):(Active:  3m51s,  %  non-child:  100.00%)
                          -  BytesReceived:  24.00  B
                          -  ConvertRowBatchTime:  1.301us
                          -  DataArrivalWaitTime:  487.306ms
                          -  DeserializeRowBatchTimer:  5.929us
                          -  FirstBatchArrivalWaitTime:  487.305ms
                          -  PeakMemoryUsage:  4.01  KB
                          -  RowsReturned:  1
                          -  RowsReturnedRate:  0
                          -  SendersBlockedTotalTimer(*):  0ns
        Fragment  1:
            Instance  f9255570a33e421c-9072a386f7446d63  (host=TNetworkAddress(hostname:172.18.65.3,  port:9060)):(Active:  486.524ms,  %  non-child:  0.00%)
                  -  FragmentCpuTime:  343.947ms
                  -  MemoryLimit:  2.00  GB
                  -  PeakMemoryUsage:  48.00  KB
                  -  PeakReservation:  0.00  
                  -  PeakUsedReservation:  0.00  
                  -  RowsProduced:  1
                BlockMgr:
                      -  BlockWritesOutstanding:  0
                      -  BlocksCreated:  0
                      -  BlocksRecycled:  0
                      -  BufferedPins:  0
                      -  BytesWritten:  0.00  
                      -  MaxBlockSize:  8.00  MB
                      -  TotalBufferWaitTime:  0ns
                      -  TotalEncryptionTime:  0ns
                      -  TotalIntegrityCheckTime:  0ns
                      -  TotalReadBlockTime:  0ns
                DataStreamSender  (dst_id=2,  dst_fragments=[f9255570a33e421c-9072a386f7446d64]):(Active:  64.252us,  %  non-child:  0.00%)
                      -  BytesSent:  24.00  B
                      -  IgnoreRows:  0
                      -  LocalBytesSent:  0.00  
                      -  OverallThroughput:  364.7744140625  KB/sec
                      -  PeakMemoryUsage:  8.00  KB
                      -  SerializeBatchTime:  4.410us
                      -  UncompressedRowBatchSize:  24.00  B
                AGGREGATION_NODE  (id=1):(Active:  486.193ms,  %  non-child:  0.14%)
                      -  Probe  Method:  HashTable  Linear  Probing
                      -  BuildTime:  319.561ms
                      -  GetResultsTime:  0ns
                      -  HTResize:  0
                      -  HTResizeTime:  0ns
                      -  HashBuckets:  0
                      -  HashCollisions:  0
                      -  HashFailedProbe:  0
                      -  HashFilledBuckets:  0
                      -  HashProbe:  0
                      -  HashTravelLength:  0
                      -  LargestPartitionPercent:  0
                      -  MaxPartitionLevel:  0
                      -  NumRepartitions:  0
                      -  PartitionsCreated:  0
                      -  PeakMemoryUsage:  32.00  KB
                      -  RowsProcessed:  0
                      -  RowsRepartitioned:  0
                      -  RowsReturned:  1
                      -  RowsReturnedRate:  2
                      -  SpilledPartitions:  0
                    OLAP_SCAN_NODE  (id=0):(Active:  163.472ms,  %  non-child:  0.07%)
                          -  BatchQueueWaitTime:  144.422ms
                          -  BytesRead:  73.75  MB
                          -  NumDiskAccess:  1
                          -  NumScanners:  2
                          -  PeakMemoryUsage:  0.00  
                          -  RowsRead:  9.518352M  (9518352)
                          -  RowsReturned:  9.518352M  (9518352)
                          -  RowsReturnedRate:  58.226159M  /sec
                          -  ScannerBatchWaitTime:  475.513ms
                          -  ScannerWorkerWaitTime:  6.594ms
                          -  TabletCount  :  2
                          -  TotalReadThroughput:  0.0  /sec
                        OlapScanner(ods_count_advertising_log):
                              -  BlockConvertTime:  87.893ms
                              -  BlockFetchTime:  222.764ms
                              -  ReaderInitTime:  6.167ms
                              -  RowsDelFiltered:  0
                              -  RowsPushedCondFiltered:  0
                              -  ScanCpuTime:  835.740ms
                              -  ScanTime:  804.1ms
                              -  ShowHintsTime_V1:  0ns
                            SegmentIterator:
                                  -  BitmapIndexFilterTimer:  4.800us
                                  -  BlockLoadTime:  127.20ms
                                  -  BlockSeekCount:  85
                                  -  BlockSeekTime:  566.317us
                                  -  BlocksLoad:  9.298K  (9298)
                                  -  CachedPagesNum:  1.173K  (1173)
                                  -  CompressedBytesRead:  0.00  
                                  -  DecompressorTimer:  0ns
                                  -  IOTimer:  0ns
                                  -  IndexLoadTime_V1:  0ns
                                  -  NumSegmentFiltered:  3
                                  -  NumSegmentTotal:  8
                                  -  RawRowsRead:  9.518352M  (9518352)
                                  -  RowsBitmapIndexFiltered:  0
                                  -  RowsBloomFilterFiltered:  0
                                  -  RowsConditionsFiltered:  0
                                  -  RowsKeyRangeFiltered:  26.584791M  (26584791)
                                  -  RowsStatsFiltered:  26.566656M  (26566656)
                                  -  RowsVectorPredFiltered:  0
                                  -  TotalPagesNum:  1.173K  (1173)
                                  -  UncompressedBytesRead:  0.00  
                                  -  VectorPredEvalTime:  44.359ms
            Instance  f9255570a33e421c-9072a386f7446d5f  (host=TNetworkAddress(hostname:172.18.65.8,  port:9060)):
                  -  FragmentCpuTime:  0ns
                  -  MemoryLimit:  2.00  GB
                  -  PeakMemoryUsage:  48.00  KB
                  -  PeakReservation:  0.00  
                  -  PeakUsedReservation:  0.00  
                  -  RowsProduced:  0
                BlockMgr:
                      -  BlockWritesOutstanding:  0
                      -  BlocksCreated:  0
                      -  BlocksRecycled:  0
                      -  BufferedPins:  0
                      -  BytesWritten:  0.00  
                      -  MaxBlockSize:  8.00  MB
                      -  TotalBufferWaitTime:  0ns
                      -  TotalEncryptionTime:  0ns
                      -  TotalIntegrityCheckTime:  0ns
                      -  TotalReadBlockTime:  0ns
                DataStreamSender  (dst_id=2,  dst_fragments=[f9255570a33e421c-9072a386f7446d64]):(Active:  17.567us,  %  non-child:  0.00%)
                      -  BytesSent:  0.00  
                      -  IgnoreRows:  0
                      -  LocalBytesSent:  0.00  
                      -  OverallThroughput:  0.0  /sec
                      -  PeakMemoryUsage:  8.00  KB
                      -  SerializeBatchTime:  0ns
                      -  UncompressedRowBatchSize:  0.00  
                AGGREGATION_NODE  (id=1):(Active:  138.672us,  %  non-child:  0.00%)
                      -  Probe  Method:  HashTable  Linear  Probing
                      -  BuildTime:  261.120ms
                      -  GetResultsTime:  0ns
                      -  HTResize:  0
                      -  HTResizeTime:  0ns
                      -  HashBuckets:  0
                      -  HashCollisions:  0
                      -  HashFailedProbe:  0
                      -  HashFilledBuckets:  0
                      -  HashProbe:  0
                      -  HashTravelLength:  0
                      -  LargestPartitionPercent:  0
                      -  MaxPartitionLevel:  0
                      -  NumRepartitions:  0
                      -  PartitionsCreated:  0
                      -  PeakMemoryUsage:  32.00  KB
                      -  RowsProcessed:  0
                      -  RowsRepartitioned:  0
                      -  RowsReturned:  0
                      -  RowsReturnedRate:  0
                      -  SpilledPartitions:  0
                    OLAP_SCAN_NODE  (id=0):(Active:  200.251ms,  %  non-child:  0.09%)
                          -  BatchQueueWaitTime:  187.613ms
                          -  BytesRead:  33.35  MB
                          -  NumDiskAccess:  1
                          -  NumScanners:  2
                          -  PeakMemoryUsage:  0.00  
                          -  RowsRead:  4.303759M  (4303759)
                          -  RowsReturned:  8.266639M  (8266639)
                          -  RowsReturnedRate:  41.281262M  /sec
                          -  ScannerBatchWaitTime:  437.974ms
                          -  ScannerWorkerWaitTime:  180.298ms
                          -  TabletCount  :  2
                          -  TotalReadThroughput:  0.0  /sec
                        OlapScanner(ods_count_advertising_log):
                              -  BlockConvertTime:  35.739ms
                              -  BlockFetchTime:  94.518ms
                              -  ReaderInitTime:  2.827ms
                              -  RowsDelFiltered:  0
                              -  RowsPushedCondFiltered:  0
                              -  ScanCpuTime:  715.419ms
                              -  ScanTime:  689.400ms
                              -  ShowHintsTime_V1:  0ns
                            SegmentIterator:
                                  -  BitmapIndexFilterTimer:  2.370us
                                  -  BlockLoadTime:  55.214ms
                                  -  BlockSeekCount:  40
                                  -  BlockSeekTime:  318.328us
                                  -  BlocksLoad:  4.204K  (4204)
                                  -  CachedPagesNum:  530
                                  -  CompressedBytesRead:  0.00  
                                  -  DecompressorTimer:  0ns
                                  -  IOTimer:  0ns
                                  -  IndexLoadTime_V1:  0ns
                                  -  NumSegmentFiltered:  1
                                  -  NumSegmentTotal:  4
                                  -  RawRowsRead:  8.268687M  (8268687)
                                  -  RowsBitmapIndexFiltered:  0
                                  -  RowsBloomFilterFiltered:  0
                                  -  RowsConditionsFiltered:  0
                                  -  RowsKeyRangeFiltered:  16.509837M  (16509837)
                                  -  RowsStatsFiltered:  16.50688M  (16506880)
                                  -  RowsVectorPredFiltered:  0
                                  -  TotalPagesNum:  530
                                  -  UncompressedBytesRead:  0.00  
                                  -  VectorPredEvalTime:  19.98ms
            Instance  f9255570a33e421c-9072a386f7446d60  (host=TNetworkAddress(hostname:172.18.65.7,  port:9060)):
                  -  FragmentCpuTime:  0ns
                  -  MemoryLimit:  2.00  GB
                  -  PeakMemoryUsage:  48.00  KB
                  -  PeakReservation:  0.00  
                  -  PeakUsedReservation:  0.00  
                  -  RowsProduced:  0
                BlockMgr:
                      -  BlockWritesOutstanding:  0
                      -  BlocksCreated:  0
                      -  BlocksRecycled:  0
                      -  BufferedPins:  0
                      -  BytesWritten:  0.00  
                      -  MaxBlockSize:  8.00  MB
                      -  TotalBufferWaitTime:  0ns
                      -  TotalEncryptionTime:  0ns
                      -  TotalIntegrityCheckTime:  0ns
                      -  TotalReadBlockTime:  0ns
                DataStreamSender  (dst_id=2,  dst_fragments=[f9255570a33e421c-9072a386f7446d64]):(Active:  19.651us,  %  non-child:  0.00%)
                      -  BytesSent:  0.00  
                      -  IgnoreRows:  0
                      -  LocalBytesSent:  0.00  
                      -  OverallThroughput:  0.0  /sec
                      -  PeakMemoryUsage:  8.00  KB
                      -  SerializeBatchTime:  0ns
                      -  UncompressedRowBatchSize:  0.00  
                AGGREGATION_NODE  (id=1):(Active:  123.667us,  %  non-child:  0.00%)
                      -  Probe  Method:  HashTable  Linear  Probing
                      -  BuildTime:  388.806ms
                      -  GetResultsTime:  0ns
                      -  HTResize:  0
                      -  HTResizeTime:  0ns
                      -  HashBuckets:  0
                      -  HashCollisions:  0
                      -  HashFailedProbe:  0
                      -  HashFilledBuckets:  0
                      -  HashProbe:  0
                      -  HashTravelLength:  0
                      -  LargestPartitionPercent:  0
                      -  MaxPartitionLevel:  0
                      -  NumRepartitions:  0
                      -  PartitionsCreated:  0
                      -  PeakMemoryUsage:  32.00  KB
                      -  RowsProcessed:  0
                      -  RowsRepartitioned:  0
                      -  RowsReturned:  0
                      -  RowsReturnedRate:  0
                      -  SpilledPartitions:  0
                    OLAP_SCAN_NODE  (id=0):(Active:  474.188ms,  %  non-child:  0.20%)
                          -  BatchQueueWaitTime:  455.611ms
                          -  BytesRead:  48.53  MB
                          -  NumDiskAccess:  1
                          -  NumScanners:  2
                          -  PeakMemoryUsage:  0.00  
                          -  RowsRead:  6.262978M  (6262978)
                          -  RowsReturned:  12.357826M  (12357826)
                          -  RowsReturnedRate:  26.061025M  /sec
                          -  ScannerBatchWaitTime:  839.75ms
                          -  ScannerWorkerWaitTime:  542.461ms
                          -  TabletCount  :  2
                          -  TotalReadThroughput:  0.0  /sec
                        OlapScanner(ods_count_advertising_log):
                              -  BlockConvertTime:  58.606ms
                              -  BlockFetchTime:  147.142ms
                              -  ReaderInitTime:  2.933ms
                              -  RowsDelFiltered:  0
                              -  RowsPushedCondFiltered:  0
                              -  ScanCpuTime:  1s119ms
                              -  ScanTime:  1s81ms
                              -  ShowHintsTime_V1:  0ns
                            SegmentIterator:
                                  -  BitmapIndexFilterTimer:  2.637us
                                  -  BlockLoadTime:  83.100ms
                                  -  BlockSeekCount:  37
                                  -  BlockSeekTime:  257.49us
                                  -  BlocksLoad:  6.117K  (6117)
                                  -  CachedPagesNum:  768
                                  -  CompressedBytesRead:  0.00  
                                  -  DecompressorTimer:  0ns
                                  -  IOTimer:  0ns
                                  -  IndexLoadTime_V1:  0ns
                                  -  NumSegmentFiltered:  2
                                  -  NumSegmentTotal:  4
                                  -  RawRowsRead:  12.358192M  (12358192)
                                  -  RowsBitmapIndexFiltered:  0
                                  -  RowsBloomFilterFiltered:  0
                                  -  RowsConditionsFiltered:  0
                                  -  RowsKeyRangeFiltered:  6.407552M  (6407552)
                                  -  RowsStatsFiltered:  6.397952M  (6397952)
                                  -  RowsVectorPredFiltered:  0
                                  -  TotalPagesNum:  768
                                  -  UncompressedBytesRead:  0.00  
                                  -  VectorPredEvalTime:  30.102ms
            Instance  f9255570a33e421c-9072a386f7446d61  (host=TNetworkAddress(hostname:172.18.65.6,  port:9060)):
                  -  FragmentCpuTime:  0ns
                  -  MemoryLimit:  2.00  GB
                  -  PeakMemoryUsage:  48.00  KB
                  -  PeakReservation:  0.00  
                  -  PeakUsedReservation:  0.00  
                  -  RowsProduced:  0
                BlockMgr:
                      -  BlockWritesOutstanding:  0
                      -  BlocksCreated:  0
                      -  BlocksRecycled:  0
                      -  BufferedPins:  0
                      -  BytesWritten:  0.00  
                      -  MaxBlockSize:  8.00  MB
                      -  TotalBufferWaitTime:  0ns
                      -  TotalEncryptionTime:  0ns
                      -  TotalIntegrityCheckTime:  0ns
                      -  TotalReadBlockTime:  0ns
                DataStreamSender  (dst_id=2,  dst_fragments=[f9255570a33e421c-9072a386f7446d64]):(Active:  13.613us,  %  non-child:  0.00%)
                      -  BytesSent:  0.00  
                      -  IgnoreRows:  0
                      -  LocalBytesSent:  0.00  
                      -  OverallThroughput:  0.0  /sec
                      -  PeakMemoryUsage:  8.00  KB
                      -  SerializeBatchTime:  0ns
                      -  UncompressedRowBatchSize:  0.00  
                AGGREGATION_NODE  (id=1):(Active:  93.790us,  %  non-child:  0.00%)
                      -  Probe  Method:  HashTable  Linear  Probing
                      -  BuildTime:  231.764ms
                      -  GetResultsTime:  0ns
                      -  HTResize:  0
                      -  HTResizeTime:  0ns
                      -  HashBuckets:  0
                      -  HashCollisions:  0
                      -  HashFailedProbe:  0
                      -  HashFilledBuckets:  0
                      -  HashProbe:  0
                      -  HashTravelLength:  0
                      -  LargestPartitionPercent:  0
                      -  MaxPartitionLevel:  0
                      -  NumRepartitions:  0
                      -  PartitionsCreated:  0
                      -  PeakMemoryUsage:  32.00  KB
                      -  RowsProcessed:  0
                      -  RowsRepartitioned:  0
                      -  RowsReturned:  0
                      -  RowsReturnedRate:  0
                      -  SpilledPartitions:  0
                    OLAP_SCAN_NODE  (id=0):(Active:  358.653ms,  %  non-child:  0.15%)
                          -  BatchQueueWaitTime:  347.442ms
                          -  BytesRead:  27.67  MB
                          -  NumDiskAccess:  1
                          -  NumScanners:  2
                          -  PeakMemoryUsage:  0.00  
                          -  RowsRead:  3.571264M  (3571264)
                          -  RowsReturned:  7.173696M  (7173696)
                          -  RowsReturnedRate:  20.001744M  /sec
                          -  ScannerBatchWaitTime:  573.310ms
                          -  ScannerWorkerWaitTime:  588.490ms
                          -  TabletCount  :  2
                          -  TotalReadThroughput:  0.0  /sec
                        OlapScanner(ods_count_advertising_log):
                              -  BlockConvertTime:  29.890ms
                              -  BlockFetchTime:  77.242ms
                              -  ReaderInitTime:  3.896ms
                              -  RowsDelFiltered:  0
                              -  RowsPushedCondFiltered:  0
                              -  ScanCpuTime:  563.364ms
                              -  ScanTime:  543.156ms
                              -  ShowHintsTime_V1:  0ns
                            SegmentIterator:
                                  -  BitmapIndexFilterTimer:  1.777us
                                  -  BlockLoadTime:  44.457ms
                                  -  BlockSeekCount:  40
                                  -  BlockSeekTime:  170.439us
                                  -  BlocksLoad:  3.489K  (3489)
                                  -  CachedPagesNum:  440
                                  -  CompressedBytesRead:  0.00  
                                  -  DecompressorTimer:  0ns
                                  -  IOTimer:  0ns
                                  -  IndexLoadTime_V1:  0ns
                                  -  NumSegmentFiltered:  2
                                  -  NumSegmentTotal:  4
                                  -  RawRowsRead:  7.175744M  (7175744)
                                  -  RowsBitmapIndexFiltered:  0
                                  -  RowsBloomFilterFiltered:  0
                                  -  RowsConditionsFiltered:  0
                                  -  RowsKeyRangeFiltered:  1.666983M  (1666983)
                                  -  RowsStatsFiltered:  1.654784M  (1654784)
                                  -  RowsVectorPredFiltered:  0
                                  -  TotalPagesNum:  440
                                  -  UncompressedBytesRead:  0.00  
                                  -  VectorPredEvalTime:  15.810ms
            Instance  f9255570a33e421c-9072a386f7446d62  (host=TNetworkAddress(hostname:172.18.65.4,  port:9060)):
                  -  FragmentCpuTime:  0ns
                  -  MemoryLimit:  2.00  GB
                  -  PeakMemoryUsage:  48.00  KB
                  -  PeakReservation:  0.00  
                  -  PeakUsedReservation:  0.00  
                  -  RowsProduced:  0
                BlockMgr:
                      -  BlockWritesOutstanding:  0
                      -  BlocksCreated:  0
                      -  BlocksRecycled:  0
                      -  BufferedPins:  0
                      -  BytesWritten:  0.00  
                      -  MaxBlockSize:  8.00  MB
                      -  TotalBufferWaitTime:  0ns
                      -  TotalEncryptionTime:  0ns
                      -  TotalIntegrityCheckTime:  0ns
                      -  TotalReadBlockTime:  0ns
                DataStreamSender  (dst_id=2,  dst_fragments=[f9255570a33e421c-9072a386f7446d64]):(Active:  14.672us,  %  non-child:  0.00%)
                      -  BytesSent:  0.00  
                      -  IgnoreRows:  0
                      -  LocalBytesSent:  0.00  
                      -  OverallThroughput:  0.0  /sec
                      -  PeakMemoryUsage:  8.00  KB
                      -  SerializeBatchTime:  0ns
                      -  UncompressedRowBatchSize:  0.00  
                AGGREGATION_NODE  (id=1):(Active:  86.91us,  %  non-child:  0.00%)
                      -  Probe  Method:  HashTable  Linear  Probing
                      -  BuildTime:  159.341ms
                      -  GetResultsTime:  0ns
                      -  HTResize:  0
                      -  HTResizeTime:  0ns
                      -  HashBuckets:  0
                      -  HashCollisions:  0
                      -  HashFailedProbe:  0
                      -  HashFilledBuckets:  0
                      -  HashProbe:  0
                      -  HashTravelLength:  0
                      -  LargestPartitionPercent:  0
                      -  MaxPartitionLevel:  0
                      -  NumRepartitions:  0
                      -  PartitionsCreated:  0
                      -  PeakMemoryUsage:  32.00  KB
                      -  RowsProcessed:  0
                      -  RowsRepartitioned:  0
                      -  RowsReturned:  0
                      -  RowsReturnedRate:  0
                      -  SpilledPartitions:  0
                    OLAP_SCAN_NODE  (id=0):(Active:  281.374ms,  %  non-child:  0.12%)
                          -  BatchQueueWaitTime:  274.27ms
                          -  BytesRead:  20.73  MB
                          -  NumDiskAccess:  1
                          -  NumScanners:  2
                          -  PeakMemoryUsage:  0.00  
                          -  RowsRead:  2.675513M  (2675513)
                          -  RowsReturned:  5.327673M  (5327673)
                          -  RowsReturnedRate:  18.934436M  /sec
                          -  ScannerBatchWaitTime:  436.42ms
                          -  ScannerWorkerWaitTime:  427.408ms
                          -  TabletCount  :  2
                          -  TotalReadThroughput:  0.0  /sec
                        OlapScanner(ods_count_advertising_log):
                              -  BlockConvertTime:  22.401ms
                              -  BlockFetchTime:  58.173ms
                              -  ReaderInitTime:  2.644ms
                              -  RowsDelFiltered:  0
                              -  RowsPushedCondFiltered:  0
                              -  ScanCpuTime:  447.554ms
                              -  ScanTime:  432.737ms
                              -  ShowHintsTime_V1:  0ns
                            SegmentIterator:
                                  -  BitmapIndexFilterTimer:  944ns
                                  -  BlockLoadTime:  33.570ms
                                  -  BlockSeekCount:  39
                                  -  BlockSeekTime:  175.447us
                                  -  BlocksLoad:  2.614K  (2614)
                                  -  CachedPagesNum:  332
                                  -  CompressedBytesRead:  0.00  
                                  -  DecompressorTimer:  0ns
                                  -  IOTimer:  0ns
                                  -  IndexLoadTime_V1:  0ns
                                  -  NumSegmentFiltered:  1
                                  -  NumSegmentTotal:  3
                                  -  RawRowsRead:  5.329721M  (5329721)
                                  -  RowsBitmapIndexFiltered:  0
                                  -  RowsBloomFilterFiltered:  0
                                  -  RowsConditionsFiltered:  0
                                  -  RowsKeyRangeFiltered:  6.108181M  (6108181)
                                  -  RowsStatsFiltered:  6.094848M  (6094848)
                                  -  RowsVectorPredFiltered:  0
                                  -  TotalPagesNum:  332
                                  -  UncompressedBytesRead:  0.00  
                                  -  VectorPredEvalTime:  12.59ms`

**Comment by larryju** [comment]
看线程状态是卡在了循环里面，

Thread 12 (Thread 0x7f0a9338d700 (LWP 31217)):
#0  0x00007f0b3dfcf965 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib64/libpthread.so.0
#1  0x000000000389913c in std::condition_variable::wait(std::unique_lock<std::mutex>&) ()
#2  0x0000000001a67e39 in doris::OlapScanNode::transfer_thread(doris::RuntimeState*) () at ../src/exec/olap_scan_node.cpp:1458
#3  0x0000000003907400 in execute_native_thread_routine ()
#4  0x00007f0b3dfcbdd5 in start_thread () from /lib64/libpthread.so.0
#5  0x00007f0b3d9f302d in clone () from /lib64/libc.so.6

**Comment by larryju** [comment]
`while (UNLIKELY(_running_thread == assigned_thread_num && _scan_row_batches.empty() &&
                            !_scanner_done)) {
                SCOPED_TIMER(_scanner_wait_batch_timer);
                _scan_batch_added_cv.wait(l);
            }`

**Comment by dataroaring** [comment]
hi, could you describe data volume, doris version, table create sql statement?

**Reply by larryju** [reply]
`CREATE TABLE `xxxxx` (
  `id` bigint(20) NULL COMMENT "",
  `app_id` varchar(100) NULL COMMENT "",
  `date` date NULL COMMENT "",
  `user_id` bigint(20) NULL COMMENT "",
  `a` varchar(100) NULL COMMENT "",
  `b` bigint(20) NULL COMMENT "",
  `c` varchar(100) NULL COMMENT "",
  `d` varchar(100) NULL COMMENT "",
  `e` varchar(100) NULL COMMENT "",
  `f` varchar(100) NULL COMMENT "",
  `g` varchar(100) NULL COMMENT "",
  `h` varchar(100) NULL COMMENT "",
  `i` varchar(100) NULL COMMENT "",
  `j` varchar(100) NULL COMMENT "",
  `k` varchar(100) NULL COMMENT "",
  `l` bigint(20) NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`id`, `app_id`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`channel_code`) BUCKETS 10
PROPERTIES (
"replication_allocation" = "tag.location.default: 3",
"in_memory" = "false",
"storage_format" = "V2"
);`

**Reply by larryju** [reply]
doris version : 0.15.0

**Reply by larryju** [reply]
![image](https://user-images.githubusercontent.com/18693798/159400751-87a877b6-d8bd-4066-9a20-15672d1a16d8.png)

**Comment by hf200012** [comment]
@larryju You add me on WeChat, I will help you locate and solve the problem, my WeChat: 35926237
